### ISSUE TITLE
Travis CI hit by attrs breakage

### ISSUE THREAD

**Issue by dmtucker** [dbader/pytest-mypy#55]
https://travis-ci.org/dbader/pytest-mypy/jobs/602328209#L1267
```
1252Traceback (most recent call last):
1253  File "/home/travis/build/dbader/pytest-mypy/.tox/py35-pytest4.0-mypy0.570/bin/py.test", line 11, in <module>
1254    sys.exit(main())
1255  File "/home/travis/build/dbader/pytest-mypy/.tox/py35-pytest4.0-mypy0.570/lib/python3.5/site-packages/_pytest/config/__init__.py", line 58, in main
1256    config = _prepareconfig(args, plugins)
1257  File "/home/travis/build/dbader/pytest-mypy/.tox/py35-pytest4.0-mypy0.570/lib/python3.5/site-packages/_pytest/config/__init__.py", line 182, in _prepareconfig
1258    config = get_config()
1259  File "/home/travis/build/dbader/pytest-mypy/.tox/py35-pytest4.0-mypy0.570/lib/python3.5/site-packages/_pytest/config/__init__.py", line 153, in get_config
1260    pluginmanager.import_plugin(spec)
1261  File "/home/travis/build/dbader/pytest-mypy/.tox/py35-pytest4.0-mypy0.570/lib/python3.5/site-packages/_pytest/config/__init__.py", line 522, in import_plugin
1262    __import__(importspec)
1263  File "/home/travis/build/dbader/pytest-mypy/.tox/py35-pytest4.0-mypy0.570/lib/python3.5/site-packages/_pytest/tmpdir.py", line 25, in <module>
1264    class TempPathFactory(object):
1265  File "/home/travis/build/dbader/pytest-mypy/.tox/py35-pytest4.0-mypy0.570/lib/python3.5/site-packages/_pytest/tmpdir.py", line 35, in TempPathFactory
1266    lambda p: Path(os.path.abspath(six.text_type(p)))
1267TypeError: attrib() got an unexpected keyword argument 'convert'
```
https://github.com/pytest-dev/pytest/issues/5900

**Comment by dmtucker** [comment]
This appears to affect pytest3.4 also:
```
py38-pytest3.4 create: /Users/davidmtucker/Projects/pytest-mypy/.tox/py38-pytest3.4
py38-pytest3.4 installdeps: pytest ~= 3.4.0, pytest-cov ~= 2.5.1, pytest-randomly ~= 2.1.1
py38-pytest3.4 inst: /Users/davidmtucker/Projects/pytest-mypy/.tox/.tmp/package/22/pytest-mypy-0.4.1.tar.gz
py38-pytest3.4 installed: attrs==19.3.0,coverage==4.5.4,mypy==0.740,mypy-extensions==0.4.3,pluggy==0.6.0,py==1.8.0,pytest==3.4.2,pytest-cov==2.5.1,pytest-mypy==0.4.1,pytest-randomly==2.1.1,six==1.12.0,typed-ast==1.4.0,typing-extensions==3.7.4.1
py38-pytest3.4 run-test-pre: PYTHONHASHSEED='4080983515'
py38-pytest3.4 run-test: commands[0] | py.test --cov pytest_mypy --cov-fail-under 100 --cov-report term-missing
Traceback (most recent call last):
  File "/Users/davidmtucker/Projects/pytest-mypy/.tox/py38-pytest3.4/bin/py.test", line 6, in <module>
    from pytest import main
  File "/Users/davidmtucker/Projects/pytest-mypy/.tox/py38-pytest3.4/lib/python3.8/site-packages/pytest.py", line 13, in <module>
    from _pytest.fixtures import fixture, yield_fixture
  File "/Users/davidmtucker/Projects/pytest-mypy/.tox/py38-pytest3.4/lib/python3.8/site-packages/_pytest/fixtures.py", line 845, in <module>
    class FixtureFunctionMarker(object):
  File "/Users/davidmtucker/Projects/pytest-mypy/.tox/py38-pytest3.4/lib/python3.8/site-packages/_pytest/fixtures.py", line 847, in FixtureFunctionMarker
    params = attr.ib(convert=attr.converters.optional(tuple))
TypeError: attrib() got an unexpected keyword argument 'convert'
ERROR: InvocationError for command /Users/davidmtucker/Projects/pytest-mypy/.tox/py38-pytest3.4/bin/py.test --cov pytest_mypy --cov-fail-under 100 --cov-report term-missing (exited with code 1)
_________________________________________________________________ summary _________________________________________________________________
ERROR:   py34-pytest3.4: commands failed
ERROR:   py35-pytest3.4: commands failed
ERROR:   py36-pytest3.4: commands failed
ERROR:   py37-pytest3.4: commands failed
ERROR:   py38-pytest3.4: commands failed
```
Not sure why this wasn't hit before 🤔

**Comment by dmtucker** [comment]
> Not sure why this wasn't hit before

Pytest 3.4 is neither 3.0 nor 3.x, so it wasn't checked.

**Comment by dmtucker** [comment]
This affects 3.3 too: https://github.com/pytest-dev/pytest/commit/07b2b18a01a3ba12757a4eb1c0e082922f5299e5

```
py38-pytest3.3 create: /Users/davidmtucker/Projects/pytest-mypy/.tox/py38-pytest3.3
py38-pytest3.3 installdeps: pytest ~= 3.3.0, pytest-cov ~= 2.5.1, pytest-randomly ~= 2.1.1
py38-pytest3.3 inst: /Users/davidmtucker/Projects/pytest-mypy/.tox/.tmp/package/22/pytest-mypy-0.4.1.tar.gz
py38-pytest3.3 installed: attrs==19.3.0,coverage==4.5.4,mypy==0.740,mypy-extensions==0.4.3,pluggy==0.6.0,py==1.8.0,pytest==3.3.2,pytest-cov==2.5.1,pytest-mypy==0.4.1,pytest-randomly==2.1.1,six==1.12.0,typed-ast==1.4.0,typing-extensions==3.7.4.1
py38-pytest3.3 run-test-pre: PYTHONHASHSEED='3906030418'
py38-pytest3.3 run-test: commands[0] | py.test --cov pytest_mypy --cov-fail-under 100 --cov-report term-missing
Traceback (most recent call last):
  File "/Users/davidmtucker/Projects/pytest-mypy/.tox/py38-pytest3.3/bin/py.test", line 6, in <module>
    from pytest import main
  File "/Users/davidmtucker/Projects/pytest-mypy/.tox/py38-pytest3.3/lib/python3.8/site-packages/pytest.py", line 13, in <module>
    from _pytest.fixtures import fixture, yield_fixture
  File "/Users/davidmtucker/Projects/pytest-mypy/.tox/py38-pytest3.3/lib/python3.8/site-packages/_pytest/fixtures.py", line 842, in <module>
    class FixtureFunctionMarker(object):
  File "/Users/davidmtucker/Projects/pytest-mypy/.tox/py38-pytest3.3/lib/python3.8/site-packages/_pytest/fixtures.py", line 844, in FixtureFunctionMarker
    params = attr.ib(convert=attr.converters.optional(tuple))
TypeError: attrib() got an unexpected keyword argument 'convert'
ERROR: InvocationError for command /Users/davidmtucker/Projects/pytest-mypy/.tox/py38-pytest3.3/bin/py.test --cov pytest_mypy --cov-fail-under 100 --cov-report term-missing (exited with code 1)
```

The issue was fixed in 3.5: https://github.com/pytest-dev/pytest/commit/82cdc487cefbbbd1848705427fed6276461835e6

However, it was reintroduced in 4.0.1: https://github.com/pytest-dev/pytest/commit/fc61bdd9073d86819eb54c8603384a10c626a9d4
- This is why the 3.4 traceback is different from the original.

Then it was finally fixed in 4.3.0: https://github.com/pytest-dev/pytest/commit/40cec637d7b9faf7f3794cd6c71517f98704c817
